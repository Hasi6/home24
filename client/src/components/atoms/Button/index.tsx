import styled from 'styled-components';

export const Button = styled.button`
  padding: 10px;
  border: none;
  background-color: ${({ theme }) => theme.colors.platinum600};
  color: white;
  cursor: pointer;
  border-radius: 6px;

  &:hover {
    background-color: ${({ theme }) => theme.colors.platinum800};
  }
`;
