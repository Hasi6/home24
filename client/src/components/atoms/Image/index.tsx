import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 150px;
  object-fit: cover;

  @media ${(props) => props.theme.media.sm} {
    height: 200px;
  }

  @media ${(props) => props.theme.media.md} {
    height: 250px;
  }
`;
