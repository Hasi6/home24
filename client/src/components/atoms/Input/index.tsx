import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';

type Props = InputHTMLAttributes<HTMLInputElement>;

const StyledInput = styled.input`
  padding: 8px 15px;
  border: 2px solid #ccc;
  border-radius: 5px;
  font-size: 16px;
  color: #333;
  background-color: #fff;

  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);

  ::placeholder {
    color: #aaa;
  }

  &:focus {
    outline: none;
    border-color: #0070f3;
    box-shadow: 0 2px 5px rgba(0, 112, 243, 0.2);
  }

  &:disabled {
    background-color: #f5f5f5;
    color: #bbb;
  }
`;

export function Input(props: Props) {
  return <StyledInput type='text' {...props} />;
}
