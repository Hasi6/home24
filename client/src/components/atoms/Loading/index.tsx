import styled from 'styled-components';

export const Loading = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  font-size: 24px;
  color: #666;
`;
