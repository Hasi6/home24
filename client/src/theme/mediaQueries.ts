import breakpoints from '@/theme/breakpoints';

const mediaQueries = {
  xs: `(min-width: ${breakpoints.xs})`,
  sm: `(min-width: ${breakpoints.sm})`,
  md: `(min-width: ${breakpoints.md})`,
  lg: `(min-width: ${breakpoints.lg})`,
  xl: `(min-width: ${breakpoints.xl})`
};

export default mediaQueries;
