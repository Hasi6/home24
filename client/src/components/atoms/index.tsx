import { Button } from '@/components/atoms/Button';
import { Image } from '@/components/atoms/Image';
import { Text } from '@/components/atoms/Text';
import { Loading } from '@/components/atoms/Loading';
import { Input } from '@/components/atoms/Input';

export { Button, Image, Text, Loading, Input };
