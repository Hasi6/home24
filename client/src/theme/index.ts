import breakpoints from '@/theme/breakpoints';
import mediaQueries from '@/theme/mediaQueries';

export const theme = {
  colors: {
    primary: '#0070f3',
    background: '#fff',
    text: '#333',
    white: '#fff',
    platinum500: '#e3e3e3',
    platinum600: '#b6b6b6',
    platinum800: '#5b5b5b',
    grey800: '#232322'
  },
  breakpoints,
  media: mediaQueries
};
