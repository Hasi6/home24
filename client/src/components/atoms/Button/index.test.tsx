import React from 'react';
import 'jest-styled-components';

import { Button } from '@/components/atoms';
import { render } from '@/utils/test/render';
import { theme } from '@/theme/index';

describe('Button component', () => {
  it('renders correctly', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has correct initial styles', () => {
    const { getByText } = render(<Button>Click me</Button>);
    const button = getByText('Click me');
    expect(button).toHaveStyleRule('padding', '10px');
    expect(button).toHaveStyleRule('background-color', theme.colors.platinum600);
    expect(button).toHaveStyleRule('color', 'white');
    expect(button).toHaveStyleRule('cursor', 'pointer');
  });

  it('changes background on hover', () => {
    const { getByText } = render(<Button>Click me</Button>);
    const button = getByText('Click me');
    expect(button).toHaveStyleRule('background-color', theme.colors.platinum800, {
      modifier: ':hover'
    });
  });
});
