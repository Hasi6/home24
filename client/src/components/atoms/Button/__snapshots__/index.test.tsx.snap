// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
.c0 {
  padding: 10px;
  border: none;
  background-color: #b6b6b6;
  color: white;
  cursor: pointer;
  border-radius: 6px;
}

.c0:hover {
  background-color: #5b5b5b;
}

<button
  class="c0"
/>
`;
